### PR TITLE
libvirt: README updated for Fedora 33+ systemd-resolved

### PR DESCRIPTION
**Pull request by timhansendev**
Fedora 34 uses systemd-resolved, which adds an additional step for properly resolving to the dnsmasq process for the openshift installer.

Signed-off-by: Tim Hansen <tihansen@redhat.com>

### CODE REVIEW

File: docs/dev/libvirt/README.md [hunk 1]
```diff
@@ -247,6 +247,31 @@ This step allows installer and users to resolve cluster-internal hostnames from
...
+
+Fedora 34 switched to utilizing systemd-resolved and changes how DNS resolution works on a per-interface basis. 
+
+If you are running Fedora 34, you must indicate that you want your resolver to point to the local dnsmasq resolver on the `tt0` interface.
+
+The below example uses the default `tt0` interface, change the interface name if you are utilizing a custom interface/bridge instead.
+
+```sh
+resolvectl dns tt0 127.0.0.1
+```
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:260]: I wonder if on f33+, it would be workable to not setup dnsmasq at all on the host, and make use of https://github.com/openshift/installer/commit/a6ef6c7d0418ba81ecf6e88d9bcc6e65e5ac6ca7 instead (+ maybe some `resolvectl domain tt0 ~tt.testing` configuration?)

File: docs/dev/libvirt/README.md [hunk 2]
```diff
@@ -247,6 +247,31 @@ This step allows installer and users to resolve cluster-internal hostnames from
 
 3. Reload NetworkManager to pick up the `dns` configuration change: `sudo systemctl reload NetworkManager`
 
+### Setting the resolver on Fedora 34+ with systemd-resolved
+
+Fedora 34 switched to utilizing systemd-resolved and changes how DNS resolution works on a per-interface basis. 
+
+If you are running Fedora 34, you must indicate that you want your resolver to point to the local dnsmasq resolver on the `tt0` interface.
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:254]: I find this sentence a bit confusing, we want to configure systemd-resolved to use the dnsmasq resolver we configured before for the `tt0` interface rather than the one libvirt set up automatically for this interface.

File: docs/dev/libvirt/README.md [hunk 3]
```diff
@@ -247,6 +247,31 @@ This step allows installer and users to resolve cluster-internal hostnames from
...
+
+The below example uses the default `tt0` interface, change the interface name if you are utilizing a custom interface/bridge instead.
+
+```sh
+resolvectl dns tt0 127.0.0.1
+```
+
+You may see which interfaces point to which DNS resolvers by typing `resolvectl`. `tt0` should be now pointing to 127.0.0.1, where the dnsmasq process is listening.
+
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:263]: Does this configuration survives a destroy/create sequence (when the libvirt network is destroyed/recreated)?

File: docs/dev/libvirt/README.md [hunk 4]
```diff
@@ -247,6 +247,31 @@ This step allows installer and users to resolve cluster-internal hostnames from
 
 3. Reload NetworkManager to pick up the `dns` configuration change: `sudo systemctl reload NetworkManager`
 
+### Setting the resolver on Fedora 34+ with systemd-resolved
+
```
[the remark this answers is not in the document]
Comment (timhansendev) [reply docs/dev/libvirt/README.md:251]: Noted, will change.

File: docs/dev/libvirt/README.md (hunk 1 above)
Comment (timhansendev) [reply docs/dev/libvirt/README.md:260]: I did not notice this section before. Thanks for pointing it out- I will test with this solution as this is cleaner than setting it up the way I've proposed.

File: docs/dev/libvirt/README.md (hunk 3 above)
Comment (timhansendev) [reply docs/dev/libvirt/README.md:263]: It does not, this needs to be re-created every time.  I will most likely change this section favoring the method outlined in your comment above https://github.com/openshift/installer/pull/4929#discussion_r633562305

File: docs/dev/libvirt/README.md (hunk 2 above)
Comment (timhansendev) [reply docs/dev/libvirt/README.md:254]: Noted, I will rephrase this to make it clearer as to what is actually happening.

File: docs/dev/libvirt/README.md [hunk 5]
```diff
@@ -247,11 +247,11 @@ This step allows installer and users to resolve cluster-internal hostnames from
 
 3. Reload NetworkManager to pick up the `dns` configuration change: `sudo systemctl reload NetworkManager`
 
-### Setting the resolver on Fedora 34+ with systemd-resolved
+### Setting the resolver on Fedora 33+ with systemd-resolved
```
Comment (cfergeau) [inline docs/dev/libvirt/README.md:250]: :+1:

File: docs/dev/libvirt/README.md (hunk 3 above)
Comment (cfergeau) [reply docs/dev/libvirt/README.md:263]: Did you have a chance to try https://github.com/openshift/installer/pull/4929#discussion_r633562305 ?
If not/if it did not work, we can go with this PR, but mentioning that this change is not persistent would be useful.

**Review by cfergeau — COMMENTED** [review]
The f33 commit could be squashed into the main one.

File: docs/dev/libvirt/README.md (hunk 3 above)
Comment (timhansendev) [reply docs/dev/libvirt/README.md:263]: I would like to address this in a separate PR to simplify how networking is done for libvirt development setups.  I was not able to get a working setup without dnsmasq running on the host. I think there could be some discussion on what the best architecture would be to make sure it is more automated in terms of deployment.

**Comment by timhansendev** [comment]
> The f33 commit could be squashed into the main one.

Done. Please let me know if you want to see any other fixes. Thank you

File: docs/dev/libvirt/README.md [hunk 6]
```diff
@@ -247,6 +247,34 @@ This step allows installer and users to resolve cluster-internal hostnames from
...
+...
+Link 100 (tt0)
+    Current Scopes: none
+         Protocols: -DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
+Current DNS Server: 127.0.0.1
+       DNS Servers: 127.0.0.1
+...
+```
+
+Please note that this is not persistant across libvirt network creation/deletion of the tt0 interface. Once the tt0 interface is deleted and recreated, it will need to updated again with the proper resolver.
```
Comment (timhansendev) [inline docs/dev/libvirt/README.md:275]: Added the note here that this does not persist across network creation/deletion.

Comment (cfergeau) [reply docs/dev/libvirt/README.md:275]: I've attempted to consolidate this PR and https://github.com/openshift/installer/pull/4564 in https://github.com/openshift/installer/pull/5132 so that we can finally merge this.

Comment (timhansendev) [reply docs/dev/libvirt/README.md:275]: Thanks, do you want me to close this PR in favor of #5132 ?

**Comment by timhansendev** [comment]
Work has moved towards #4564
